tests/publiccloud/check_registercloudguest.pm

```diff
@@ -93,19 +93,22 @@ sub run {
     if (is_byos()) {
         $instance->run_ssh_command(cmd => 'sudo SUSEConnect --version');
         $instance->run_ssh_command(cmd => "sudo SUSEConnect $regcode_param");
-        $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest --clean");
+        # The registercloudguest tool is not yet part of 15-SP5 as the infrastructure is not yet ready for it.
+        $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest --clean") if (get_var('PUBLIC_CLOUD_QAM'));
     }
 
-    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param");
+    # The registercloudguest tool is not yet part of 15-SP5 as the infrastructure is not yet ready for it.
+    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param") if (get_var('PUBLIC_CLOUD_QAM'));
     if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0) {
         die('The list of zypper repositories is empty.');
     }
     if ($instance->run_ssh_command(cmd => 'sudo ls /etc/zypp/credentials.d/* | wc -l') == 0) {
         die('Directory /etc/zypp/credentials.d/ is empty.');
     }
 
-    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param --force-new");
-    if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0) {
+    # The registercloudguest tool is not yet part of 15-SP5 as the infrastructure is not yet ready for it.
+    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param --force-new") if (get_var('PUBLIC_CLOUD_QAM'));
+    if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0 and !get_var('PUBLIC_CLOUD_QAM')) {
```

Comment:
this way it will run in SLE 15 SP5 but just not die in case of failure . I would just simply take all commands in one condition : 
```
if (get_var('PUBLIC_CLOUD_QAM')) {
 $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param --force-new") 
    if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0) {
        die('The list of zypper repositories is empty.');
    }
    if ($instance->run_ssh_command(cmd => 'sudo ls /etc/zypp/credentials.d/* | wc -l') == 0) {
        die('Directory /etc/zypp/credentials.d/ is empty.');
    }
}
```